### DISCUSSION TITLE
[API Request] Add a "LISTENED" state for podcasts and set it when a podcast is scrobbled

### DISCUSSION THREAD

**Discussion by lxea** [opensubsonic/open-subsonic-api#117] [API requests]
### Type of change

API extension

### Proposal description

Podcast episodes currently can have the following states: `NEW, DOWNLOADING, COMPLETED, ERROR, DELETED, SKIPPED` and I propose adding a new `LISTENED` state also.

Adding a new state to indicate that a podcast episode has been listened to would be useful for syncing state between clients.

The state can be set as listened by re-using the scrobble endpoint and if the scrobbled id is a podcast episode ID it can be marked as "LISTENED"

### Backward compatibility impact

This should only be an additive change. However some clients may need to be updated to make use of the scrobble endpoint upon podcast completion.

### Backward compatibility

- [X] No backward compatibility impact.

### API details

The [/scrobble](https://opensubsonic.netlify.app/docs/endpoints/scrobble/) endpoint should be called with the podcast ID and `submission` set to true (as is default) 

Podcast episodes when listed with `getPodcasts&includeEpisodes=true` if they have been scrobbled will now have the option of the status field being set as `listened`

**Comment by paulijar** [comment]
I think this does have some backward compatibility impact. If the server returns a new enumerated value in an old property, then it's not certain that all legacy clients can tolerate that.

How about if, instead, we would add the property `playCount` to the podcast episodes? This would be analogous with the song object.

**Reply by sentriz** [reply]
some playCount make sense to me. since it exists in other places too. a new enum value makes it looks like maybe the LISTENED state is shared across different users

**Reply by lachlan-00** [reply]
if playcount isn't there it should be added. then just a listened bool for played by me

```
playCount: int
listened: true/false
```

**Reply by Tolriq** [reply]
What does listened add here ? If play count > 0 then it's listened.
But lastPlayed for full consistency would make sense no ?

**Reply by lxea** [reply]
Would playcount be a global playcount?

**Reply by sentriz** [reply]
i think per-user making the request makes sense. that's how playCount is for songs/albums already on some servers

**Reply by lxea** [reply]
Okay cool so we want a playCount and lastPlayed field and both are user specific?

I'll update the main discussion body in a bit :)

**Reply by tamland** [reply]
`playCount` is already implemented in the original subsonic server and airsonic-advanced at least. Perhaps it's just undocumented?
